usagov/usagov-2021
# USAGOV-2016-v2: Redirect on bad characters in URL

Redirect when there's a non-breaking space at the end of a request path


https://cm-jira.usa.gov/browse/USAGOV-2016

## Description
Updated the nginx configs to redirect the user away from a URL that ends in a space/thin-space/similar to the same URL without it.
While I was able to test this in the CMS configuration, this is a blind update to the WWW nginx configs, as I am not sure how to test/run the WWW docker container on my local? This test mainly needs to be done on the WWW configs.

## Type of Changes


## Testing Instructions
* Going to a URL of `/foo/bar` should load a not-found page, which is fine
* but after merging this, going to `/foo/bar%C2%A0` should redirect you to `/foo/bar`
* and so should `/foo/bar%E2%80%89` and `/foo/bar%E2%80%83`
* mainly I want to make sure that this configuration is now crashing the WWW nginx container on Dev since this is kind of a blind edit there as I dont quite know how to run the WWW container on my local

## Reviewer Reminders

- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions

Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
2. Find the commit of this pull request.
3. Build and deploy the changes.
4. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.
